### PR TITLE
Pin home07 new clv

### PR DESCRIPTION
**Pull request by wafakd**
Add new CLV for PinnedNews

### CODE REVIEW

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl [hunk 1]
```diff
@@ -0,0 +1,194 @@
+<%
+    import javax.jcr.Node;
+    import org.exoplatform.wcm.webui.paginator.UICustomizeablePaginator;
```
[the remark this answers is not in the document]
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl]: ok

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+%>
+
+<script>
+
+	require(['SHARED/jquery', 'SHARED/dotdotdot'], function (jQuery, dotdotdot) {
+		jQuery(document).ready(function () {
+			jQuery(".contentTitle").dotdotdot();
+			jQuery(".contentBodyText").dotdotdot();
+			jQuery(".rightTitle").dotdotdot();
+			jQuery(".rightBody").dotdotdot();
```
[the remark this answers is not in the document]
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl]: its unuseful

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+		})
+	});
+</script>
+
+<%
+def header = uicomponent.getHeader();
+def isShowHeader = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_HEADER) && header != null && header.trim().length() != 0;
```
[the remark this answers is not in the document]
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl]: ok

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+<div id="OfficeMiddleActualites">
+
+	<div class="news-header">
+		<div class="global-title">
+			<h3 class="news-title-top">Actualités</h3>
+		</div>
+
+		<div class="read-all-news">
+			<a class="see-all-news" onclick="window.open('/portal/intranet/news');">See all news</a>
+		</div>
```
[the remark this answers is not in the document]
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl]: ok, i will remove it.

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+	}
+}
+
+        def itemName = viewNode.getName();
+        def itemTitle = uicomponent.getTitle(viewNode);
+          if (itemTitle != null && itemTitle.trim().length() != 0)
+            itemName = itemTitle;
+        def itemSummary = uicomponent.getSummary(viewNode);
+        def itemSummaryField = uicomponent.getSummaryField(viewNode);
+        def isShowSummary = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_SUMMARY) && itemSummary != null;
```
[the remark this answers is not in the document]
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl]: ok

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+}
+
+
+def itemName_0 = nodes.get(0).getName();
+def itemTitle_0 = uicomponent.getTitle(nodes.get(0));
+if (itemTitle_0 != null && itemTitle_0.trim().length() != 0)
+  itemName_0 = itemTitle_0;
+def itemSummary_0 = uicomponent.getSummary(nodes.get(0));
+def itemSummaryField_0 = uicomponent.getSummaryField(nodes.get(0));
+def isShowSummary_0 = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_SUMMARY) && itemSummary_0 != null;
```
[the remark this answers is not in the document]
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl]: the same for itemSummaryField_0, ok

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+%>
+		</div>
+
+		<%
+
+  }
+
+%>
+	</div>
+</div>
```
[the remark this answers is not in the document]
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl]: yes I used the code of the demo-03 and modified the import of images because it didn't work, then I added the style

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl [hunk 8]
```diff
@@ -0,0 +1,194 @@
+<%
```
[the remark this answers is not in the document]
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl]: ok

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+def header = uicomponent.getHeader();
+def isShowHeader = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_HEADER) && header != null && header.trim().length() != 0;
+
+%>
+<div id="OfficeMiddleActualites">
+
+	<div class="news-header">
+		<div class="global-title">
+			<h3 class="news-title-top">Actualités</h3>
```
[the remark this answers is not in the document]
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl]: ok fixed

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl [hunk 10]
```diff
@@ -0,0 +1,194 @@
...
+						<span>Lire la suite</span>
+						</a>
+					</span>
+					<% } %>
+
+				</div>
+			</div>
+		</div>
+		<div class="news-contentRight">
+			<% for (int i=1;i<4;i++) {
```
[the remark this answers is not in the document]
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/NewsPin2.gtmpl]: ok

File: webapp/src/main/webapp/WEB-INF/conf/news/news-templates-configuration.xml [hunk 11]
```diff
@@ -58,6 +58,23 @@
             </field>
           </object>
         </object-param>
+
+        <object-param>
+          <name>News Pin CLV2 template</name>
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/WEB-INF/conf/news/news-templates-configuration.xml]:
```suggestion
          <name>Latest News CLV template</name>
```

File: webapp/src/main/webapp/css/news.less [hunk 12]
```diff
@@ -1196,6 +1196,389 @@
...
+    color: #252525;
+    position: relative;
+  }
+
+  .news-title-top:after  {
+    content: "";
+    position: absolute;
+    width: 30px;
+    height: 4px;
+    background: #578dc9;
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/css/news.less]: Please use variable for colors and check if it matches with the eXo colors.

File: webapp/src/main/webapp/css/news.less [hunk 13]
```diff
@@ -1196,6 +1196,389 @@
...
+  font-size: 16px;
+  letter-spacing: 1px;
+  font-weight: bold;
+  padding-bottom: 10px;
+}
+
+
+
+
+
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/css/news.less]: Please remove all useless blank lines.

File: webapp/src/main/webapp/css/news.less [hunk 14]
```diff
@@ -1196,6 +1196,389 @@
...
+.contentText {
+  position: absolute;
+  bottom: 0;
+  background-image: linear-gradient(to top, rgb(0,0,0) 0%, rgba(0,0,0,0));
+  height: 180px;
+  padding: 0 20px;
+  width: 70%;
+  color: #fff;
+  transition: all 1s ease;
+}
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/css/news.less]: All the styles should be nested in a global style related to the news CLV, to avoid applying this style elsewhere in case an element with the same class name.

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl [hunk 15]
```diff
@@ -0,0 +1,162 @@
+<%
+    import org.exoplatform.wcm.webui.clv.UICLVPortlet;
+    import org.exoplatform.wcm.webui.Utils;
+    import org.exoplatform.services.wcm.core.NodeLocation;
+    import org.exoplatform.services.wcm.utils.WCMCoreUtils;
+%>
+<div id="OfficeMiddleActualites">
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: Do not use an id, since the template could potentially be displayed several times on a page, so it would result in having the same id twice in a page.

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl [hunk 16]
```diff
@@ -0,0 +1,162 @@
...
+       if(isShowHeader) {
+        %>
+		<div class="global-title">
+			<h3 class="news-title-top">$header</h3>
+		</div>
+ <%
+        }
+    %>
+		<div class="read-all-news">
+			<a class="see-all-news" onclick="#">See all news</a>
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: The labels must be internationalized.

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl [hunk 17]
```diff
@@ -0,0 +1,162 @@
...
+		</div>
+
+	</div>
+
+	<div class="news-content">
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: I think we generally use camelCase instead of hyphens for CSS classes. Can you check please and change if it is the case ?

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl [hunk 18]
```diff
@@ -0,0 +1,162 @@
...
+        nodes.add(viewNode);
+    }
+
+  def style = "";
+  def isShowTitle = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_TITLE);
+  def isShowDate = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_DATE_CREATED);
+  def isShowLink = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_LINK);
+  def isShowReadmore = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_READMORE);
+
+def itemLink_0 = uicomponent.getURL(nodes.get(0));
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: Can you rename these variables as "firstItem..." instead, to make it more clear ? For example firstItemLink, firstItemOwner, ...

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl [hunk 19]
```diff
@@ -0,0 +1,162 @@
...
+    %>
+		<div class="news-contentLeft">
+
+			<div class="backgroundLeft" style="background-image:url($imgSrc_0)">
+				<div class="overlay"></div>
+			</div>
+
+			<div class="ombre">
+			</div>
+			<!--<img class="imgContainerOption2" src="$imgSrc_0" />-->
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: To remove

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl [hunk 20]
```diff
@@ -0,0 +1,162 @@
...
+def itemLink_0 = uicomponent.getURL(nodes.get(0));
+def itemDateCreated_0 = uicomponent.getCreatedDate(nodes.get(0));
+def itemOwner_0 = uicomponent.getAuthor(nodes.get(0));
+
+def isShowImage_0 = uicomponent.isShowField(UICLVPortlet.PREFERENCE_SHOW_ILLUSTRATION);
+def imgSrc_0;
+if(isShowImage_0) {
+	if(nodes.get(0).hasNode("illustration")) {
+		def imageNode_0= nodes.get(0).getNode("illustration");
+		imgSrc_0 = WCMCoreUtils.generateImageURI(imageNode_0, null);
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: You should use `"/portal/rest/v1/news/" + node.getUUID() + "/illustration"` instead.

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl [hunk 21]
```diff
@@ -0,0 +1,162 @@
...
+%>
+		</div>
+
+		<%
+
+  }
+
+%>
+	</div>
+</div>
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: Please take care of the indentation everywhere.

File: webapp/src/main/webapp/css/news.less [hunk 22]
```diff
@@ -1196,6 +1196,389 @@
...
+  padding: 30px;
+}
+
+
+
+
+
+#secondLigneContainerHome2 .twoColumnsCLVTemplate {
+  margin: 0;
+}
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/css/news.less]: The id secondLigneContainerHome2 is not used, these blocks seem useless. Please remove all useless classes.

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl [hunk 23]
```diff
@@ -0,0 +1,162 @@
+<%
+    import org.exoplatform.wcm.webui.clv.UICLVPortlet;
+    import org.exoplatform.wcm.webui.Utils;
+    import org.exoplatform.services.wcm.core.NodeLocation;
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: Useless import.

File: webapp/src/main/webapp/WEB-INF/conf/news/news-templates-configuration.xml (hunk 11 above)
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/news-templates-configuration.xml]: ok, c mieux

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl (hunk 19 above)
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: ok

Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: ok

File: webapp/src/main/webapp/css/news.less (hunk 13 above)
Comment (wafakd) [reply webapp/src/main/webapp/css/news.less]: ok

File: webapp/src/main/webapp/css/news.less (hunk 12 above)
Comment (wafakd) [reply webapp/src/main/webapp/css/news.less]: yes its the primary color of eXo

File: services/src/main/resources/locale/portlet/news/News_fr.properties [hunk 24]
```diff
@@ -14,11 +14,11 @@ news.composer.btn.delete=Supprimer
...
 news.composer.summary=R\u00E9sum\u00E9
-news.composer.createNews=\u00C9crire un article pour {0}
+news.composer.createNews=Write an article for {0}
 news.composer.summaryDescription=Le r\u00E9sum\u00E9 appara\u00EEtra dans la description de l\u2019article.
-news.composer.placeholderTitleInput=Titre
-news.composer.placeholderSummaryInput=R\u00E9sum\u00E9
-news.composer.placeholderContentInput=Contenu
+news.composer.placeholderTitleInput=Title
+news.composer.placeholderSummaryInput=Summary
+news.composer.placeholderContentInput=Content
```
Comment (thomasdelhomenie) [inline services/src/main/resources/locale/portlet/news/News_fr.properties]: These changes must be reverted.

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl (hunk 23 above)
Comment (thomasdelhomenie) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: Comment not considered.

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl (hunk 15 above)
Comment (thomasdelhomenie) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: Comment not considered.

File: webapp/src/main/webapp/css/news.less (hunk 12 above)
Comment (thomasdelhomenie) [reply webapp/src/main/webapp/css/news.less]: ok. Please add a variable.

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl [hunk 25]
```diff
@@ -0,0 +1,224 @@
...
+        %>
+		<div class="global-title">
+			<h3 class="news-title-top">$header</h3>
+		</div>
+ <%
+        }
+    %>
+	</div>
+
+	<div class="news-content">
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: We should use a better name IMO, this one is too generic. It should be something like `latestNews`. WDYT ?

File: webapp/src/main/webapp/css/news.less [hunk 26]
```diff
@@ -1238,3 +1420,125 @@
...
+.notification_space{
+  font-weight: bold;
+}
+.notification_settings_url {
+  margin:30px 0 10px 0;
+  color: #2f5e92;
+  text-decoration: none;
+  font-size:13px;
+  font-family:HelveticaNeue,arial,tahoma,serif
+}
```
Comment (thomasdelhomenie) [inline webapp/src/main/webapp/css/news.less]: Styles related to notifications must not be part of this PR.

**Comment by thomasdelhomenie** [comment]
@wafakd I did some tests, here my feedbacks:
* weird effect when mousing over the main news : the text goes down a bit before going up
* bad positioning with long summary


* when no articles to display, the portlet should not display anything
* labels are not internationalized (for example "Lire la suite") 
* I should be able to open the news by clicking anywhere on the news (illustration, summary)
* in mobile view, no space between the first news and the second news

File: webapp/src/main/webapp/css/news.less (hunk 12 above)
Comment (wafakd) [reply webapp/src/main/webapp/css/news.less]: ok

File: webapp/src/main/webapp/css/news.less (hunk 14 above)
Comment (wafakd) [reply webapp/src/main/webapp/css/news.less]: ok

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl (hunk 15 above)
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: i will remove and replace it by the class name

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl (hunk 21 above)
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: done

File: webapp/src/main/webapp/css/news.less (hunk 22 above)
Comment (wafakd) [reply webapp/src/main/webapp/css/news.less]: done

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl (hunk 25 above)
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: yes it's a good idea

File: webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl (hunk 23 above)
Comment (wafakd) [reply webapp/src/main/webapp/WEB-INF/conf/news/application-templates/content-list-viewer/list/LatestNews.gtmpl]: ok done

**Comment by thomasdelhomenie** [comment]
Replaced by https://github.com/exo-addons/news/pull/95
